vulture_os/toolkit/api_parser/beyondtrust_reportings/beyondtrust_reportings.py

```diff
@@ -0,0 +1,325 @@
+
+Vulture OS is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "Nicolas Lançon"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'Beyondtrust Reportings API'
+__parser__ = 'Beyondtrust Reportings'
+
+from json import dumps as json_dumps
+import logging
+
+import requests
+from xmltodict import parse as xmltodict_parse
+
+from datetime import datetime, timedelta
+from django.utils import timezone
+from django.conf import settings
+
+from toolkit.api_parser.api_parser import ApiParser
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class BeyondtrustReportingsAPIError(Exception):
+    pass
+
+
+class BeyondtrustReportingsParser(ApiParser):
+    HEADERS = {
+        "Content-Type": "application/json",
+        'Accept': 'application/json'
+    }
+
+    DURATION = 86400  # Fetch 24h of logs (in seconds)
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.beyondtrust_reportings_client_id = data["beyondtrust_reportings_client_id"]
+        self.beyondtrust_reportings_secret = data["beyondtrust_reportings_secret"]
+        self.beyondtrust_reportings_host = data["beyondtrust_reportings_host"]
+        self.beyondtrust_reportings_api_token = data.get("beyondtrust_reportings_api_token", {})
+
+        self.allowed_reporting_types = {
+            "Team": data.get("beyondtrust_reportings_get_team_logs", False),
+            "AccessSession": data.get("beyondtrust_reportings_get_access_session_logs", False),
+            "VaultAccountActivity": data.get("beyondtrust_reportings_get_vault_account_activity_logs", False),
+            "SupportSession": data.get("beyondtrust_reportings_get_support_session_logs", False),
+        }
+
+        if not self.beyondtrust_reportings_host.startswith('https://') and not self.beyondtrust_reportings_host.startswith('http://'):
+            self.beyondtrust_reportings_host = f"https://{self.beyondtrust_reportings_host}"
+        self.beyondtrust_reportings_host = self.beyondtrust_reportings_host.rstrip("/")
+
+        self.session = None
+
+    def save_access_token(self):
+        try:
+            if self.frontend:
+                self.frontend.beyondtrust_reportings_api_token = self.beyondtrust_reportings_api_token
+                self.frontend.save()
```

Comment:
```suggestion
```
Not necessary: is done once and for all at each finishing run (even when raising)